kong/api/routes/debug.lua

```diff
@@ -0,0 +1,102 @@
+local get_sys_filter_level = require("ngx.errlog").get_sys_filter_level
+local set_log_level = require("resty.kong.log").set_log_level
+
+local LOG_LEVELS = require("kong.constants").LOG_LEVELS
+
+local ngx = ngx
+local kong = kong
+local pcall = pcall
+local type = type
+local tostring = tostring
+
+local function handle_put_log_level(self, broadcast)
+  if kong.configuration.database == "off" then
+    local message = "cannot change log level when not using a database"
+    return kong.response.exit(405, { message = message })
+  end
+
+  local log_level = LOG_LEVELS[self.params.log_level]
+
+  if type(log_level) ~= "number" then
+    return kong.response.exit(400, { message = "unknown log level" })
+  end
+
+  local sys_filter_level = get_sys_filter_level()
+
+  if sys_filter_level and sys_filter_level == log_level then
+    local message = "log level is already " .. self.params.log_level
+    return kong.response.exit(200, { message = message })
+  end
+
+  local ok, err = pcall(set_log_level, log_level)
+
+  if not ok then
+    local message = "failed setting log level: " .. tostring(err)
+    return kong.response.exit(500, { message = message })
+  end
+
+  -- broadcast to all workers in a node
+  ok, err = kong.worker_events.post("debug", "log_level", log_level)
+
+  if not ok then
+    local message = "failed broadcasting to workers: " .. err
+    return kong.response.exit(500, { message = message })
+  end
+
+  if broadcast then
+    -- broadcast to all nodes in a cluster
+    ok, err = kong.cluster_events:broadcast("log_level", tostring(log_level))
+
+    if not ok then
+      local message = "failed broadcasting to cluster: " .. err
+      return kong.response.exit(500, { message = message })
+    end
+  end
+
+  -- store in shm so that newly spawned workers can update their log levels
+  ok, err = ngx.shared.kong_log_level:set("level", log_level)
+
+  if not ok then
+    local message = "failed storing log level in shm: " .. tostring(err)
+    return kong.response.exit(500, { message = message })
+  end
+
+  return kong.response.exit(200, { message = "log level changed" })
+end
+
+local routes = {
+  ["/debug/node/log-level"] = {
+    GET = function(self)
+      local sys_filter_level = get_sys_filter_level()
+      local cur_level = LOG_LEVELS[sys_filter_level]
+
+      if type(cur_level) ~= "string" then
+        local message = "unknown log level: " .. tostring(sys_filter_level)
+        return kong.response.exit(500, { message = message })
+      end
+
+      return kong.response.exit(200, { message = "log level: " .. cur_level })
```

Comment:
is it to return a structured result that `cur_level` is the value? i'm thinking of something like
`{ "data": { "message": "log level is now set to <cur_level>", "log-level": "<cur_level>"} }` so
it's easier to parse and expect. how do you think?

Reply:
@fffonion I like your idea, could you please add this suggestion to the Design Doc? Changes to API design/contract/signature need to be discussed there

Reply:
I have a theory that we're going to be redesigning this API when we add in data plane configuration. I'm guessing the API will begin accepting input like
```
{
  "log_level": "debug",
  "nodes": ["dataplane1", "dataplane2"],
  "timeout": 60
}
```
And I think at that point it also makes sense to return a structured response about what nodes had their log level updated

Reply:
if it's explictly write out (as i do see it in design doc), let's keep it as is for now. as Tyler mentioned
this may got new fields post config scoping, we can add our change there. i'm also adding this
 to design doc as well, but consider this not a blocker.